src/cli/miner_cli_genesis.erl

```diff
@@ -322,6 +325,135 @@ genesis_proof(["genesis", "proof", PrivKeyB58], [], []) ->
 genesis_proof(_, [], []) ->
     usage.
 
+%%
+%% genesis recreate
+%%
+
+genesis_recreate_cmd() ->
+    [
+        [["genesis", "recreate", '*', '*'], [], [], fun genesis_recreate/3]
+    ].
+
+genesis_recreate_usage() ->
+    [["genesis", "recreate"],
+        ["genesis recreate <pubkey> <privkey>\n\n",
+            "  recreate a genesis block from the existing ledger data\n"
+            "  includingtxns for all chainvars, gateways, accounts and validators.\n"
+            "  pub and priv key should be base64url encoded binary terms.\n"
```

Comment:
Are the keys the 'master key' for the chain?

Reply:
Would suggest supplying an output directory or file here, vs putting it in whatever the VM's working directory is.

Reply:
yeah keys here are the master keys for the chain.  pubkey is then set as 'master_key' in the var txn and the priv key used to create the proof

Reply:
NOTE: i simplified the key handling...requirement now is to supply the master key itself as a base58 encoded string rather than its pub and priv components